File: lab-1/filldeck.rb
#The number of possible types for each card attribute.
NUM_TYPES = 3

#Fills Set Deck with all cards needed to play one game.
def fillDeck
    deck = []

    #Create each card so that each card is unique and there is a combination of every possible type of card.
    for a in 1..NUM_TYPES do
        for b in 1..NUM_TYPES do
            for c in 1..NUM_TYPES do
                for d in 1..NUM_TYPES do
                    deck << Card.new(a,b,c,d)
                end
            end
        end
    end

    return deck
end

File: lab-2/js/fillDeck.js
//The number of possible types for each card attribute
var NUM_TYPES = 3;
// Fills Set deck with all cards needed to play one game.
function fillDeck(deck) {

    //Create each card so that each card is unique and there is a combination of every possible type of card.
    for (var a = 1; a <= NUM_TYPES; a++) {
        for (var b = 1; b <= NUM_TYPES; b++) {
            for (var c = 1; c <= NUM_TYPES; c++) {
                for (var d = 1; d <= NUM_TYPES; d++) {
                    deck.push(new Card(a,b,c,d));
                }
            }
        }
    }

    return deck;
}
